sftp/client.go

```diff
@@ -111,3 +126,12 @@ func (c *clientImpl) Remove(path string) error {
 func (c *clientImpl) MkdirAll(path string) error {
 	return c.client.MkdirAll(path)
 }
+
+func (c *clientImpl) Reset() error {
+	newSFTPClient, err := newSFTPClientFromConfig(c.config)
+	if err != nil {
+		return err
+	}
+	c.client = newSFTPClient
```

Comment:
I wonder if this could happen concurrently? It looks like it might preventing us from doing things like this in the filemanager:

```go
go fm.Upload(something)
go fm.Upload(somethingElse)
```

Reply:
In the calling function on the server, we have a lock on Upload, so we are good here. However, we can add a lock here as well to prevent any unwanted flow
https://github.com/rudderlabs/rudder-server/blob/5090a8d25614626a36633a7cf4bb5958897c0b3a/router/batchrouter/handle_async.go#L336

Reply:
Yes, I'd say we either add a lock here as well or a comment on top of the `FileManager` constructor maybe saying that this isn't concurrent safe.

Reply:
Added the comment for now.